File: 4.Clients/WebApi/ClientApp/src/entities/offer-stage.ts
import { StageStatusEnum } from './enums/stage-status.enum';
import { SeniorityEnum } from './enums/seniority.enum';
import { Stage } from './stage';
import { Offer } from './offer';

export class OfferStage extends Stage {    
    hireDate: Date;
    seniority: SeniorityEnum;    
    backgroundCheckDone: boolean;
    backgroundCheckDoneDate: Date;
    preocupationalDone: boolean;
    preocupationalDoneDate: Date;
    offers : Offer[];
}

File: 1.Domain/Domain.Model/OfferStage.cs
using Domain.Model.Enum;
using System;
using System.Collections.Generic;
using System.Text;

namespace Domain.Model
{
    public class OfferStage: Stage
    {        
        public DateTime HireDate { get; set; }
        public Seniority Seniority { get; set; }      
        public bool BackgroundCheckDone { get; set; }
        public DateTime? BackgroundCheckDoneDate { get; set; }
        public bool PreocupationalDone { get; set; }
        public DateTime? PreocupationalDoneDate { get; set; }
        public List<Offer> Offers { get; set; }
    }
}
